### PR TITLE
Upgrade some versions + sonar version from 9.3 to 9.8

### PR DESCRIPTION
**Pull request by oussamaLaribi**

### CODE REVIEW

**Comment by dedece35** [comment]
Hi @oussamaLaribi,
i've just made a review with some questions. please see them.
another question / note : 
- do you run tests locally ? unit tests and manual tests with integration of plugin in a old SonarQube like a 9.3 version ? (to verify compatibility)
- I've just corrected a mistake in build github worklfow on main branch (during repository migration, I disabled build&analyse phase because of sonarQube secrets not ready) : I've just enabled again a build&test phase in build github workflow, I relaunch the worklow for your branch to check my correction with your branch, but your banch is not up-to-date ... could you upgrade your branch and commit again something to run the workflow automatically please ?

**Comment by oussamaLaribi** [comment]
> Hi @oussamaLaribi, i've just made a review with some questions. please see them. another question / note :
> 
> * do you run tests locally ? unit tests and manual tests with integration of plugin in a old SonarQube like a 9.3 version ? (to verify compatibility)
> * I've just corrected a mistake in build github worklfow on main branch (during repository migration, I disabled build&analyse phase because of sonarQube secrets not ready) : I've just enabled again a build&test phase in build github workflow, I relaunch the worklow for your branch to check my correction with your branch, but your banch is not up-to-date ... could you upgrade your branch and commit again something to run the workflow automatically please ?
> * next step will be to report this sonarQube upgrade to ecocode-mobile, please

Hi @dedece35, thanks for the review.
Yes, I did those local tests.
I upgraded my branch also. It's up to date now.
Next, I will report this sonarQube upgrade to ecocode-mobile.

**Comment by dedece35** [comment]
@oussamaLaribi ok for new workflow with build&test phase.
but you don't take into account my 3 pending review comments 😜 : please see above.

**Comment by oussamaLaribi** [comment]
> @oussamaLaribi ok for new workflow with build&test phase. but you don't take into account my 3 pending review comments 😜 : please see above.

@dedece35 I can't find these pending review comments

**Comment by dedece35** [comment]
@oussamaLaribi, sorry, my bad, I forget to submit my review :p : done.
now, if SonarQube 9.4 is required, I think you have to mention it in installation documentation.

File: pom.xml [hunk 1]
```diff
@@ -167,6 +167,11 @@
 			<artifactId>sonar-plugin-api</artifactId>
 			<scope>compile</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.sonarsource.sonarqube</groupId>
+			<artifactId>sonar-plugin-api-impl</artifactId>
+			<scope>compile</scope>
```
[the remark this answers is not in the document]
Comment (oussamaLaribi) [reply pom.xml:173]: When I delete the scope I got "package org.sonar.api.internal does not exist" I don't know why.

File: python-plugin/pom.xml [hunk 2]
```diff
@@ -39,6 +39,7 @@
         <dependency>
             <groupId>org.sonarsource.sonarqube</groupId>
             <artifactId>sonar-plugin-api-impl</artifactId>
+            <scope>compile</scope>
```
[the remark this answers is not in the document]
Comment (oussamaLaribi) [reply python-plugin/pom.xml:42]: The same thing, when I delete the scope I got "package org.sonar.api.internal does not exist" I don't know why.

**Comment by oussamaLaribi** [comment]
> @oussamaLaribi, sorry, my bad, I forget to submit my review :p : done. now, if SonarQube 9.4 is required, I think you have to mention it in installation documentation.

@dedece35 Okay, I'll mention it. Thanks

File: python-plugin/src/main/java/fr/cnumr/python/RulesList.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
+/*
```
[the remark this answers is not in the document]
Comment (dedece35) [reply python-plugin/src/main/java/fr/cnumr/python/RulesList.java:1]: ... and java :p

File: pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dedece35) [reply pom.xml:173]: I will do tests to check the problem i you want

File: python-plugin/src/main/java/fr/cnumr/python/RulesList.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (oussamaLaribi) [reply python-plugin/src/main/java/fr/cnumr/python/RulesList.java:1]: In java, we already have externalizing.

File: pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (oussamaLaribi) [reply pom.xml:173]: Okay, thanks.

File: python-plugin/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (oussamaLaribi) [reply python-plugin/pom.xml:42]: Okay, thanks.

**Review by dedece35 — APPROVED** [review]

**Comment by dedece35** [comment]
Hi @oussamaLaribi ,
I made few minutes ago a refactoring / optimization on `pom.xml` files : there were some mistakes on scopes on dependencies.
Please, update your branch with my modifications. I think you will have some conflicts, be carefull.

Then you can delete "<scope>compile</scope>" because it is the default scope on your dependency `sonar-plugin-api-impl`

**Comment by oussamaLaribi** [comment]
> Hi @oussamaLaribi , I made few minutes ago a refactoring / optimization on `pom.xml` files : there were some mistakes on scopes on dependencies. Please, update your branch with my modifications. I think you will have some conflicts, be carefull.
> 
> Then you can delete "compile" because it is the default scope on your dependency `sonar-plugin-api-impl`

Hi @dedece35, Okay I'll do it. Thanks.

**Comment by dedece35** [comment]
@oussamaLaribi, I can help if you need. tell me if you want some help.

**Comment by dedece35** [comment]
hi @oussamaLaribi , do you want to say java plugin instead ? because there is compilation failure on java plugin class.
tell me if you want some help.

**Comment by oussamaLaribi** [comment]
> hi @oussamaLaribi , do you want to say java plugin instead ? because there is compilation failure on java plugin class. tell me if you want some help.

Hi @dedece35, you can check now the compilation failure on the python plugin class.

**Comment by dedece35** [comment]
@oussamaLaribi 
I made the correctino of th problem : several problems on `pom.xml` files
I made a temporary PR for you to check my modifications : please take into account all modifications on all `pom.xml` files
Problem was mainly in versions of python /php plugins.
I checked locally : 
- compilation
- package building
- deploy and execution of JAR plugins on a dockerized SonarQube

**Comment by oussamaLaribi** [comment]
> @oussamaLaribi I made the correction of the problem : several problems on `pom.xml` files I made a temporary PR #20 for you to check my modifications : please take into account all modifications on all `pom.xml` files : check commit [565dd48](https://github.com/green-code-initiative/ecoCode/commit/565dd483f368ab9ac248996ae8def5080f8b5dde) Problem was mainly in versions of python / php plugins. I checked locally :
> 
> * compilation
> * package building
> * deploy and execution of JAR plugins on a dockerized SonarQube

Hi @dedece35, Okay I'll check it. Thanks.

**Comment by dedece35** [comment]
Hi @oussamaLaribi 
is it all ok for you thus I can do again a complete review ?

**Comment by oussamaLaribi** [comment]
> Hi @oussamaLaribi is it all ok for you thus I can do again a complete review ?

Hi @dedece35 Yes, just the commented  . I can't find another way to force it, in the official doc there is nothing.

File: pom.xml [hunk 4]
```diff
@@ -80,12 +81,14 @@
 				<groupId>org.sonarsource.analyzer-commons</groupId>
 				<artifactId>sonar-analyzer-commons</artifactId>
 				<version>${sonar-analyzer-commons.version}</version>
+				<scope>compile</scope>
```
Comment (dedece35) [inline pom.xml:84]: please, delete this scope. no need to specify it. I tested locally without it and it's ok.

File: pom.xml [hunk 5]
```diff
@@ -80,12 +81,14 @@
...
 			</dependency>
 
 			<dependency>
 				<groupId>org.sonarsource.sonarqube</groupId>
 				<artifactId>sonar-plugin-api-impl</artifactId>
 				<version>${sonar-plugin-api-impl.version}</version>
+				<scope>compile</scope>
```
Comment (dedece35) [inline pom.xml:91]: please, delete this scope. no need to specify it. I tested locally without it and it's ok.

File: python-plugin/src/main/java/fr/cnumr/python/RulesList.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (dedece35) [reply python-plugin/src/main/java/fr/cnumr/python/RulesList.java:1]: ok

File: python-plugin/pom.xml [hunk 6]
```diff
@@ -63,9 +61,25 @@
...
+                    <!-- TODO : commented because plugin not loaded in SONAR if required version of python plugin is not installed -->
+                    <!-- TODO : check how to resolve it (how to force this version in SonarQube installation) -->
+                    <!-- <requirePlugins>python:${sonar.python.version}</requirePlugins> -->
+                    <basePlugin>python</basePlugin>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
+                <version>${maven-shade-plugin.version}</version>
```
Comment (dedece35) [inline python-plugin/pom.xml:73]: please move this version management to root `pom.xml` file
1 - put these lines in root `pom.xml` file (in `pluginManagement` part like other maven plugins)
`
<plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-shade-plugin</artifactId>
                <version>${maven-shade-plugin.version}</version>
</plugin>
`
2 - delete in each langage module the line with de "version"

File: python-plugin/pom.xml [hunk 7]
```diff
@@ -63,9 +61,25 @@
...
                     <sonarQubeMinVersion>${sonarqube.version}</sonarQubeMinVersion>
-                    <requirePlugins>python:${sonar.python.version}</requirePlugins>
+                    <!-- TODO : commented because plugin not loaded in SONAR if required version of python plugin is not installed -->
+                    <!-- TODO : check how to resolve it (how to force this version in SonarQube installation) -->
+                    <!-- <requirePlugins>python:${sonar.python.version}</requirePlugins> -->
+                    <basePlugin>python</basePlugin>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
```
Comment (dedece35) [inline python-plugin/pom.xml:71]: due to the use of this maven plugin, some xml files appears (like `java-plugin/dependency-reduced-pom.xml` for example), please add this line to `.gitignore` file :
`dependency-reduced-pom.xml`

**Review by dedece35 — CHANGES_REQUESTED** [review]
@oussamaLaribi , i've just made a new review. I put here somme new comments.
cheers, it's almost done :p

**Comment by oussamaLaribi** [comment]
> new review

Hi @dedece35, it's done. Thanks.

File: INSTALL.md [hunk 8]
```diff
@@ -33,6 +33,7 @@ Howto build the SonarQube ecoCode plugins
 
 - Java >= 11.0.17
 - Mvn 3
+- SonarQube 9.4 or higher
```
[the remark this answers is not in the document]
Comment (dedece35) [reply INSTALL.md]: done

File: docker-compose.yml [hunk 9]
```diff
@@ -1,7 +1,7 @@
 version: "3.3"
 services:
   sonar:
-    image: sonarqube:9.3-community
+    image: sonarqube:9.7-community
```
[the remark this answers is not in the document]
Comment (dedece35) [reply docker-compose.yml]: done

File: docker-compose.yml [hunk 10]
```diff
@@ -24,12 +24,12 @@ services:
       - type: bind
         source: ./python-plugin/target/ecocode-python-plugin-0.1.0-SNAPSHOT.jar
         target: /opt/sonarqube/extensions/plugins/ecocode-python-plugin-0.1.0-SNAPSHOT.jar
-#      - type: bind
-#        source: ./ecolinter-plugin/target/ecocode-ecolinter-plugin-0.1.0-SNAPSHOT.jar
-#        target: /opt/sonarqube/extensions/plugins/ecocode-ecolinter-plugin-0.1.0-SNAPSHOT.jar
-#      - type: bind
-#        source: ./xml-plugin/target/ecocode-xml-plugin-0.1.0-SNAPSHOT.jar
-#        target: /opt/sonarqube/extensions/plugins/ecocode-xml-plugin-0.1.0-SNAPSHOT.jar
+#     - type: bind
```
[the remark this answers is not in the document]
Comment (dedece35) [reply docker-compose.yml]: done

File: python-plugin/pom.xml (hunk 7 above)
Comment (dedece35) [reply python-plugin/pom.xml:71]: done

File: pom.xml (hunk 5 above)
Comment (dedece35) [reply pom.xml]: done

File: python-plugin/pom.xml (hunk 6 above)
Comment (dedece35) [reply python-plugin/pom.xml]: done

File: pom.xml (hunk 4 above)
Comment (dedece35) [reply pom.xml]: done

**Review by dedece35 — APPROVED** [review]
I check again and test again localy with all your last modifications.
It's now ok : you can merge your branch.

Another thing with minor priority, there is WARNINGs during build process in the console (`tool_build.sh` file execution).
After SonarQube has been started, plugins are ok regarding logs, but I think we have to find a solution to correct those Warnings during build phase. This problem is due to `maven-shade-plugin` added. Check capture above : 
Capture d’écran 2022-12-23 à 22 08 49

**Comment by dedece35** [comment]
Hi @oussamaLaribi, I've just merged your PR and check in main branch.

**Comment by oussamaLaribi** [comment]
> Hi @oussamaLaribi, I've just merged your PR and checked in main branch again (all is ok). Please report all last modifications to `ecocode-mobile` repository and make me reviewer
Hi @dedece35, Okay I'll do it. Thanks.

**Pull request merged**